src/components/structures/RoomView.js

```diff
@@ -803,6 +803,96 @@ module.exports = React.createClass({
         scrollNode.scrollTop = scrollNode.scrollHeight;
     },
 
+    // scroll the event view to put the given event at the bottom.
+    //
+    // pixel_offset gives the number of pixels between the bottom of the event
+    // and the bottom of the container.
+    scrollToEvent: function(event_id, pixel_offset) {
+        var scrollNode = this._getScrollNode();
+        if (!scrollNode) return;
+
+        var messageWrapper = this.refs.messagePanel;
+        if (messageWrapper === undefined) return;
+
+        var idx = this._indexForEventId(event_id);
+        if (idx === null) {
+            // we don't seem to have this event in our timeline. Presumably
+            // it's fallen out of scrollback. We ought to backfill until we
+            // find it, but we'd have to be careful we didn't backfill forever
+            // looking for a non-existent event.
+            //
+            // for now, just scroll to the top of the buffer.
+            console.log("Refusing to scroll to unknown event "+event_id);
+            scrollNode.scrollTop = 0;
+            return;
+        }
+
+        // we might need to roll back the messagecap (to generate tiles for
+        // older messages). Don't roll it back past the timeline we have, though.
```

Comment:
might be worth clarifying this to explain that this doesn't involve backfilling - it's just generating more tiles for events we already have in our timeline. i got a bit stuck reading the code before I realised what was going on.